File: benchmarks/fixture/go/billing.go
package fixture

import "math"

type InvoiceLine struct {
	Quantity  float64
	UnitPrice float64
}

func CalculateInvoiceTotal(lines []InvoiceLine, taxRate float64) float64 {
	var subtotal float64
	for _, line := range lines {
		subtotal += line.Quantity * line.UnitPrice
	}
	return math.Round(subtotal*(1+taxRate)*100) / 100
}

func ValidatePayment(amount float64, currency string) bool {
	return amount > 0 && len(currency) == 3
}

File: benchmarks/fixture/cpp/billing.cpp
#include <cmath>
#include <string>
#include <utility>
#include <vector>

double calculate_invoice_total(const std::vector<std::pair<double, double>>& lines, double tax_rate) {
    double subtotal = 0;
    for (const auto& [quantity, unit_price] : lines) subtotal += quantity * unit_price;
    return std::round(subtotal * (1 + tax_rate) * 100) / 100;
}

bool validate_payment(double amount, const std::string& currency) {
    return amount > 0 && currency.size() == 3;
}
